flink-runtime/src/main/java/org/apache/flink/runtime/minicluster/MiniCluster.java

```diff
@@ -669,13 +669,14 @@ public JobExecutionResult executeJobBlocking(JobGraph job) throws JobExecutionEx
 		try {
 			jobResult = jobResultFuture.get();
 		} catch (ExecutionException e) {
-			throw new JobExecutionException(job.getJobID(), "Could not retrieve JobResult.", ExceptionUtils.stripExecutionException(e));
+			throw new JobExecutionException(job.getJobID(), ApplicationStatus.UNKNOWN,
+					"Could not retrieve JobResult.", ExceptionUtils.stripExecutionException(e));
 		}
 
 		try {
 			return jobResult.toJobExecutionResult(Thread.currentThread().getContextClassLoader());
 		} catch (IOException | ClassNotFoundException e) {
-			throw new JobExecutionException(job.getJobID(), e);
+			throw new JobExecutionException(job.getJobID(), jobResult.getApplicationStatus(), e);
```

Comment:
Can this be misleading? For example the application was successful but it failed to deserialize the result so now we have status `SUCCEEDED` but with an exception.

Reply:
@kl0u IMO, I don't think this is misleading . The application status for this exception case is SUCCESS, but the exception means the deserialization of job result with an exception. We may add exception message to tell users that the application is succeeded but the deserialization with an exception. What do you think?

Reply:
Maybe this is a nice middle ground @SteNicholas . Have the status as `SUCCESS` and a message that says that something went wrong during getting the result. As a sidenote, it may be that this could require a new type of exception in the future (not necessarily in this PR) as it seems that we are using the same exception for different purposes. One is successful job execution (or not) and the other is successful job result translation (or not). 

This was also a discussion we had offline with @SteNicholas ;)